Questions for chapter 4 and 5 (#2)
willfindlay/4108final: practice.rmd

# Authentication Protocols and Key Establishment

\question{}

> \vspace{4em}

Reviewer: Delete this

> \question{Define what an Entity authentication is?}

Reviewer: ```suggestion
\question{Define entity authentication is.}
```

\vspace{8em}

> \question{What is the difference between the claimant and the verifier?}

Reviewer: more context here? claimant and verifier in <.....>

\vspace{8em}

\question{What is the difference between key transport and key agreement?}
\vspace{8em}

\question{What is an ephemeral key?}
\vspace{8em}

> \question{What is the difference between authentication only and unauthenticated key establishment?}

Reviewer: Authentication only? Do you mean authenticated and unauthenticated?

\vspace{8em}

\question{Define authenticated key establishment.}
\vspace{8em}

\question{In an initial keying what is the registration phase?}
\vspace{8em}

> \question{What is the difference between weak secret and strong secret?}

Reviewer: ```suggestion
\question{What is the difference between a weak secret and a strong secret?}
```

\vspace{8em}

\question{Describe the difference between a key distribution center (KDC) and a key translation center (KTC).}
\vspace{8em}

> \question{Alice wants to provide a proof of knowledge to Bob. Alice sends H(s) (Hash of the secret s) to Bob. Define an attack so Eve who is listening in can provided a proof of knowledge to Bob.}

Reviewer: ```suggestion
\question{Alice wants to provide a proof of knowledge to Bob. Alice sends H(s) (Hash of the secret s) to Bob. Describe an attack so Eve who is listening in can provided a proof of knowledge to Bob.}
```

\vspace{8em}

\question{Give an example of a replay attack.}
\vspace{8em}

\question{Give an example of a reflection attack.}
\vspace{8em}

\question{Give an example of a relay attack.}
\vspace{8em}

\question{Give an example of a interleaving attack.}
\vspace{8em}

\question{Give an example of a middle-person attack.}
\vspace{8em}

\question{Give an example of a dictionary attack.}
\vspace{8em}

> \question{Give an example of a forward search attack.	Give an example of a pre-capture attack.}

Reviewer: ```suggestion
\question{Give an example of a forward search attack.}
\vspace{8em}
\question{Give an example of a pre-capture attack.}
```

\vspace{8em}

> \question{Give an example of when you should use a random number, sequence number, and a timestamp.}

Reviewer: More context here

\vspace{8em}

> \question{What is the discrete log problem regarding Diffie Helman key exchange?	Describe what a small sub-group attack is.}

Reviewer: ```suggestion
\question{What is the discrete log problem with respect to Diffie-Hellman key exchange? Describe what a small sub-group attack is.}
```

\vspace{8em}

> \question{What are some properties a good key may have?}

Reviewer: This might need to be reworded

\vspace{8em}

\question{What is forward secrecy?}
\vspace{8em}

\question{What is an ephemeral key?}
\vspace{8em}

\question{What is known key secrecy?}
\vspace{8em}

\question{What does Password-authenticated key exchange (PAKE) do?}
\vspace{8em}

> \question{PAKE uses a key derivation function (KDF) to generate a symmetric key. Describe how you may recover the key.}

Reviewer: Is this from the perspective of an attacker?

\vspace{8em}

> Using DH (Diffie Helman) how can you add authentication to mitigate the risk of a man in the middle attack.}

Reviewer: ```suggestion
Using DH (Diffie-Hellman) how can you add authentication to mitigate the risk of a man in the middle attack.}
```

\vspace{8em}

> \question{Summarize the technical details of the following password-authenticated key exchange alternatives to EKE and SPEKE.

Reviewer: ```suggestion
\question{Summarize the technical details of the following password-authenticated key exchange alternatives to EKE and SPEKE.}
```

> SRP/Secure Remote Password (hint: [49, 48]).

Reviewer: ```suggestion
\subquestion{SRP/Secure Remote Password}
```

> OKE/Open Key Exchange (hint: [30], but also [31] or [9, Chapter 7]).

Reviewer: ```suggestion
\subquestion{OKE/Open Key Exchange}
```

> J-PAKE/PAKE by Juggling (hint: [21, 22, 20]).}

Reviewer: ```suggestion
\subquestion{J-PAKE/PAKE by Juggling}
```

\vspace{8em}

\question{How could we perform a forward search attack against a public key system?}
\vspace{8em}
\question{How can recognizable formats reduce the key space required to decrypt a message.}
\vspace{8em}

\question{What is a single sign on (SSO) architecture?}
\vspace{8em}

> \question{What is a single credential system? Do you need to re login when using services?}

Reviewer: ```suggestion
\question{What is a single credential system? Do you need to re-login when using services?}
```

\vspace{8em}

> \question{What dies an IdP do (identity provider).}

Reviewer: ```suggestion
\question{What does an IDP (identity provider) do?}
```

\vspace{8em}

\question{What is an authenticator?}
\vspace{8em}

\question{Explain the different types of SSO systems. Credential manager (CM), Enterprise SSO, Federal identity (FI).}
\vspace{8em}

\question{Does the Kerberos protocol provide mutual entity authentication and authenticated key establishment?}
\vspace{8em}

\question{Explain how the Kerberos protocol works.}
\vspace{8em}

> 

Reviewer: Do we need to know Kerberos?

# Operating Systems Security and Access Control

\question{Why is Multics our focus of study when looking at OS security?}
\vspace{4em}

\question{What significance does timesharing computing have with security?}
\vspace{4em}

\question{What are some goals in OS security we would like to strive for?}
\vspace{4em}

\question{What are some goals in OS security we would like to strive for?}

> \vspace{4em}

Reviewer: Duplicate. Also perhaps too general.

\question{What does the privileged bit set grant a user?}
\vspace{4em}

\question{Explain the given components of a memory protection scheme. Descriptor register, Privileged bit, and Supervisor.}
\vspace{4em}

\question{What is a limitation to the supervisor model of memory protection? }
\vspace{4em}

\question{What is the difference between a descriptor segment and segment descriptors? }
\vspace{4em}

\question{What is an access control indicator?}
\vspace{4em}

> 

Reviewer: 4em might be too small for answer space
